runtime/contract_update_validation.go

```diff
@@ -506,10 +506,14 @@ func (validator *ContractUpdateValidator) checkConformances(
 	// Having extra new conformance is OK. See: https://github.com/onflow/cadence/issues/1394
 	for _, oldConformance := range oldConformances {
 		found := false
-		for _, newConformance := range newConformances {
+		for index, newConformance := range newConformances {
 			err := oldConformance.CheckEqual(newConformance, validator)
 			if err == nil {
 				found = true
+
+				// Remove the matched conformance, so we don't have to check it again.
+				// i.e: optimization
+				newConformances = append(newConformances[:index], newConformances[index+1:]...)
```

Comment:
Great idea! Double checked with https://go.dev/play/p/N233f-oGIDh, as I was not quite sure how modifying `newConformances` here affects the `range` over it / the inner loop

Reply:
It always breaks from the inner-loop after modification, so should be safe. Thanks for double-checking!